src/main/java/io/cos/cas/osf/web/flow/login/OsfPrincipalFromNonInteractiveCredentialsAction.java

```diff
@@ -557,46 +558,55 @@ private OsfApiInstitutionAuthenticationResult notifyOsfApiOfInstnAuthnSuccess(
         }
         if (!isMemberOf.isEmpty()) {
             LOGGER.info(
-                    "[CAS XSLT] Secondary institution detected. SSO is '{}' and member is '{}'",
+                    "[CAS XSLT] Secondary institution detected: username={}, institution={}, member={}",
+                    username,
                     institutionId,
                     isMemberOf
             );
+        } else {
+            LOGGER.debug("[CAS XSLT] Secondary institution is not provided: username={}, institution={}", username, institutionId);
         }
-        final String payload = normalizedPayload.toString();
```

Comment:
The bug: `payload` was the one sent to OSF API (after encryption and encoding). `department` was added to the `normalizedPayload` after this one, which led to it not being released to OSF API. This has been fixed by moving it to after the `department` work.